### PR TITLE
Enable DPDK support on KVM

### PR DESCRIPTION
**Pull request by nvazquez**
## Description
Enable DPDK support on KVM in conjunction with Open VSwitch.

New properties can be set on `agent.properties` file to enable ports created with dpdkvhostuser type on KVM hosts:
````
openvswitch.dpdk.enabled=true
openvswitch.dpdk.ovs.path=OVS_PATH
````

DPDK guest VMs can be set by sending extra configurations as part of the 'deployVirtualMachine' or 'updateVirtualMachine' API methods. Extra configurations are controlled by a new global setting 'enable.additional.vm.configuration'.
Extra configurations are parts of XML data which are appended to the domail XML of the VM. CloudStack expects an UTF-8 URL encoded string as 'extraconfig' parameter on the mentioned API methods.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## GitHub Issue/PRs

Documentation PR: https://github.com/apache/cloudstack-documentation/pull/6 & https://github.com/apache/cloudstack-documentation/pull/8

## Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
Testing
- [ ] I have added tests to cover my changes.
- [ ] All relevant new and existing integration tests have passed.
- [ ] A full integration testsuite with all test that can run on my environment has passed.

### CODE REVIEW

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
Thanks @borisstoyanov, I've updated the PR description and also rebased master branch
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
Rebased master branch.
@blueorangutan package

**Comment by nvazquez** [comment]
Thanks @borisstoyanov, I'm seeing Jenkins failures to get the code and start checking and tried rebuilding the job but still fails with:

```> git reset --hard # timeout=10
ERROR: Error fetching remote repo 'origin'
hudson.plugins.git.GitException: Failed to fetch from https://github.com/apache/cloudstack
	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:888)
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1155)
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1186)
	at hudson.scm.SCM.checkout(SCM.java:504)
	at hudson.model.AbstractProject.checkout(AbstractProject.java:1208)
	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:574)
	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:499)
	at hudson.model.Run.execute(Run.java:1794)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
Caused by: hudson.plugins.git.GitException: Command "git reset --hard" returned status code 128:
stdout: 
stderr: fatal: unable to read tree b4862e12b63821371306b72302066bf6772d774b
````

**Comment by nvazquez** [comment]
Rebased master branch again but still getting Jenkins error trying to get the code @rhtyd @borisstoyanov

File: server/src/main/java/com/cloud/vm/UserVmManagerImpl.java [hunk 1]
```diff
@@ -514,6 +515,8 @@
     private static final ConfigKey<Boolean> AllowDeployVmIfGivenHostFails = new ConfigKey<Boolean>("Advanced", Boolean.class, "allow.deploy.vm.if.deploy.on.given.host.fails", "false",
             "allow vm to deploy on different host if vm fails to deploy on the given host ", true);
 
+    private static final ConfigKey<Boolean> EnableAdditionalVmConfig = new ConfigKey<>("Advanced", Boolean.class, "enable.additional.vm.configuration",
```
Comment (dhlaluku) [inline server/src/main/java/com/cloud/vm/UserVmManagerImpl.java]: Can we change this setting from 'enable.additional.vm.configuration' to 'additional.vm.configuration.allow' or 'additional.vm.configuration.enable'?

**Comment by rhtyd** [comment]
@nvazquez is this still in progress?

**Comment by nvazquez** [comment]
@rhtyd yes, I'll update it with latest fix once testing is completed

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@DaanHoogland @borisstoyanov latest fixes added, are we ok to merge this?

**Comment by DaanHoogland** [comment]
think we are @nvazquez

**Pull request merged**